### PR TITLE
chat color: fix usernames not being recoloured if they have an icon

### PR DESCRIPTION
**Pull request by Hydrox6**
When checking if a message's sender is your friend for username recolouring, the `ChatMessageManager` fails to account for any icon the sender might have, eg. if they're an ironman or a PMod. This PR rectifies that.

Issue initially reported by `ol long sack` in Discord.

### CODE REVIEW

File: runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java [hunk 1]
```diff
@@ -84,4 +85,67 @@ public void onChatMessage()
 
 		verify(messageNode).setValue(eq("<col=b20000>Your dodgy necklace protects you. It has <col=ff0000>1<col=b20000> charge left.</col>"));
 	}
+
+	@Test
+	public void testPublicFriendUsernameRecolouring()
+	{
+		final String localPlayerName = "RuneLite";
+		final String friendName = "Zezima";
```
Comment (makingcomment) [inline runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java:93]: Isn't Zezima a pmod (which would have an icon)?

Comment (Cyborger1) [reply runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java:93]: He's been demoted temporarily for this test. Don't worry too much about it.

Comment (Cyborger1) [reply runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java:93]: But as a more serious answer though, the names chosen here are arbitrary. Whether players exist with these names or not has no incidence on the tests.

File: runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java [hunk 2]
```diff
@@ -67,14 +68,14 @@ public void before()
...
+	public void testMessageRecoloring()
 	{
-		when(chatColorConfig.opaquePublicChat()).thenReturn(Color.decode("#b20000"));
+		when(chatColorConfig.opaqueGameMessage()).thenReturn(Color.decode("#b20000"));
 
 		chatMessageManager.loadColors();
 
 		ChatMessage chatMessage = new ChatMessage();
-		chatMessage.setType(ChatMessageType.PUBLICCHAT);
+		chatMessage.setType(ChatMessageType.CONSOLE);
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java:78]: Is the chat type for the dodgy neck message really console?

Comment (Hydrox6) [reply runelite-client/src/test/java/net/runelite/client/chat/ChatMessageManagerTest.java]: whoops, changed to GAMEMESSAGE

**Pull request merged**